File: JAVA/All sample programs (BASIC)/GCD.java
import java.io.*;
class GCD
{
static int Compute(int a,int b)
{
if(b==0)
return(a);
else
return(Compute(b,a%b));
}
public static void main(String args[])throws IOException
{
InputStreamReader read=new InputStreamReader(System.in);
BufferedReader in=new BufferedReader(read);
System.out.println("Enter the two numbers in descending order :- ");
int x=Integer.parseInt(in.readLine());
int y=Integer.parseInt(in.readLine());
int z=Compute(x,y);
System.out.println("The GCD of the two given numbers is :- "+z);
}
}

File: DS LAB SOLUTIONS FOR SEE/6/gcd.c
//gcd using recursion
#include<stdio.h>
int gcd(int m,int n)
{
	if(!n)
		return m;
	if(n>m)
		return gcd(n,m);
	return gcd(n,m%n);
}
int main()
{
	int m,n;
	printf("enter 2 no\n");
	scanf("%d%d",&n,&m);
	printf("GCD is %d\n",gcd(n,m) );
	return 0;
}